src/api/endpoint/api/package.ts

```diff
@@ -45,12 +45,14 @@ const redirectOrDownloadStream = (
   const tarballUrlRedirect = _.get(config, 'experiments.tarball_url_redirect');
   storage
     .hasLocalTarball(packageName, filename)
-    .then((hasLocalTarball) => {
+    .then(async (hasLocalTarball) => {
       if (hasLocalTarball) {
         const context = { packageName, filename };
         const tarballUrl =
           typeof tarballUrlRedirect === 'function'
-            ? tarballUrlRedirect(context)
+            ? tarballUrlRedirect.constructor.name === 'AsyncFunction'
+              ? await tarballUrlRedirect(context)
```

Comment:
If this fails, will crash the server  due there is no catch, I'm not sure about this approach mix thenable with async functions. I'm thinking how we can improve this.

Reply:
Thanks, I'll check when I have some time 🙏🏼